src/AspNetCoreModuleV2/CommonLib/application.h

```diff
@@ -30,24 +26,23 @@ class APPLICATION
     Recycle() = 0;
 
     virtual
-    ~APPLICATION();
+    ~APPLICATION() = 0 { };
```

Comment:
why not have IAPPLICATION pure abstract class and then have APPLICATION::IAPPLICATION as abstract class which can avoid duplicate code at in-process and out-of-process sub classes. Same for REQEUST_HANDLER

Reply:
There isn't too much code duplication, I would like to avoid base classes for things that ship with different schedule and live in different repos because it increases probability of us accidentaly breaking things.

Reply:
I agree with Pan here. If there's a strong reason to define an interface, you can do so with an IAPPLICATION. You should still have a base class so you don't have to implement all the methods in all derived classes.

Reply:
Common part is 3 members two of which are ref counting which we don't try to extract into common type anywhere (https://github.com/aspnet/IISIntegration/search?utf8=%E2%9C%93&q=cRefs+%3D+InterlockedDecrement%28%26m_cRefs%29%29&type=), and the third one is a trivial field access. 

We are additionally planning to split current code base into two repositories that would be built and released at a different time, I don't want accidental changes to a base class to break the product.